metadata-ingestion/src/datahub/ingestion/source/iceberg/iceberg.py

```diff
@@ -181,26 +197,20 @@ def _create_iceberg_workunit(
             aspects=[Status(removed=False)],
         )
 
-        custom_properties: Dict = dict(table.properties())
-        custom_properties["location"] = table.location()
-        try:
-            if isinstance(table, BaseTable) and table.current_snapshot():
-                custom_properties["snapshot-id"] = str(
-                    table.current_snapshot().snapshot_id
-                )
-                custom_properties[
-                    "manifest-list"
-                ] = table.current_snapshot().manifest_location
-        except KeyError:
-            # The above API is not well implemented, and can throw KeyError when there is no data.
-            pass
+        # Dataset properties aspect.
+        custom_properties = dict(table.metadata.properties)
```

Comment:
The `dict` should be redundant since `properties` is always set.
```suggestion
        custom_properties = table.metadata.properties
```

Reply:
My intent here was to make a copy of the dictionary to avoid modifying it on the next line.  I thought it could introduce unwanted side-effects later on (let's say, at the profiling stage) if I add a property to the table object.  

Maybe I should use `custom_properties = table.metadata.properties.copy()` to make my intent obvious? (No need to deepcopy since it's a `dict[str, str]`)

Reply:
+1 for `.copy()` to signal intent.